src/notifications/NotificationsManager.ts

```diff
@@ -73,37 +94,174 @@ class NotificationsManager {
   protected db: DB;
   protected keyRing: KeyRing;
   protected nodeManager: NodeManager;
+  protected taskManager: TaskManager;
   protected messageCap: number;
 
   /**
-   * Top level stores MESSAGE_COUNT_KEY -> number (of messages)
+   * Map for hashing NotificationIds to find TaskIds.
+   * The key must be NotificationIdEncoded, so that the string representation is hashed rather than the reference to the NotificationId object.
    */
-  protected notificationsDbPath: LevelPath = [this.constructor.name];
+  protected outboxNotificationIdEncodedToTaskIdMap: Map<
+    NotificationIdEncoded,
+    TaskId
+  >;
+
   /**
-   * Messages stores NotificationId -> string (message)
+   * Top level stores inbox/outbox -> Notifications, numMessages
    */
-  protected notificationsMessagesDbPath: LevelPath = [
-    this.constructor.name,
-    'messages',
+  protected notificationsManagerDbPath: LevelPath = [this.constructor.name];
+  protected notificationsManagerInboxDbPath: LevelPath = [
+    ...this.notificationsManagerDbPath,
+    'inbox',
+  ];
+  protected notificationsManagerOutboxDbPath: LevelPath = [
+    ...this.notificationsManagerDbPath,
+    'outbox',
   ];
-  protected notificationsMessageCounterDbPath: KeyPath = [
-    ...this.notificationsDbPath,
-    MESSAGE_COUNT_KEY,
+  /**
+   * Notifications stores NotificationId -> Notification
+   */
+  protected notificationsManagerInboxNotificationsDbPath: LevelPath = [
+    ...this.notificationsManagerInboxDbPath,
+    'notifications',
   ];
+  protected notificationsManagerInboxNotificationsCounterDbPath: KeyPath = [
+    ...this.notificationsManagerInboxDbPath,
+    NOTIFICATIONS_COUNT_KEY,
+  ];
```

Comment:
Is this a counter for the number of inbox messages? Don't do this because it WILL conflict without locking. Even with locking it's not a good idea. We already have a way to track the number of entries in the DB. there's a `tran.count` method that should handle this. See if you can try using that instead.

Reply:
the inbox count already existed before i worked on the notifications, i just changed the db path

Reply:
i've removed this for now